engine/v2/paymails/paymailsmodels/paymail.go

```diff
@@ -26,3 +31,8 @@ type NewPaymail struct {
 	Avatar     string
 	UserID     string
 }
+
+// CheckAvatarURL checks if avatar is either empty string or a proper url link
+func (np *NewPaymail) CheckAvatarURL() bool {
+	return np.Avatar == "" || explicitHTTPURLRegex.MatchString(np.Avatar)
```

Comment:
maybe instead of some hard to understand regex use:
```go
		URL, err := url.Parse(avatarURL)
		if err != nil {
			return ErrInvalidAvatarURL.Wrap(err)
		}

		if URL.Scheme != "http" || URL.Scheme != "https" {
           return ErrInvalidAvatarURL.Wrap(spverrors.Newf("avatarURL should have http(s) scheme"))
        }
```